sound/soc/sof/core.c

```diff
@@ -335,8 +335,8 @@ static int sof_probe(struct platform_device *pdev)
 			plat_data->machine->new_mach_data(plat_data);
 	} else {
```

Comment:
@plbossart sorry to ask a very basic question here( for my benefit)
Why do we have the new_mach_data at all? Maybe I missed something but as far as I could tell, it is only set for Baytrail to be mfld_new_mach_data and this function seems to be doing exactly when we do in the else part below. Sorry again if I missed something obvious here.

Reply:
@ranj063 it's not obvious to me either, one of those things that only Keyon and Liam understand.

Reply:
@keyonjie @lgirdwood could you please shed some light on this? Thanks!

Reply:
I think this was added by Liam to make it compatible to byt/cht sst driver, @lgirdwood can you confirm if we can discard this part and remove *new_data from struct sof_ops_table ?